src/components/MessageInput/StopGeneratingButton.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+export type StopGeneratingButtonProps = {
+  onClick: () => void;
+} & React.ComponentProps<'button'>;
+
+export const StopGeneratingButton = ({ onClick }: StopGeneratingButtonProps) => (
```

Comment:
Not sure if the "Generating" part is necessary in the naming. Looking just at the code, it is not clear what does onClick handler do.

Reply:
That's true, it can stop either "Thinking" or "Generating" on its own. The general idea is that we hide this behind a single phrase (i.e integrators don't need to know the difference between the two). What would you propose as something better ? `StopButton` is too generic I think.

Reply:
I would inline that default markup and in the component context I would call it `StopAIGenerationButton` or `StopAIOutputButton`.